dlpy/network.py

```diff
@@ -0,0 +1,1598 @@
+            print('NOTE: ' + _file_name_ + '_attr' + _extension_ +
+                  ' is used as weigths attribute.')
+            self._retrieve_('table.loadtable',
+                            caslib=cas_lib_name,
+                            path=_file_name_ + '_attr' + _extension_,
+                            casout=dict(replace=True,
+                                        name=self.model_name + '_weights_attr'))
+
+            self.set_weights_attr(self.model_name + '_weights_attr')
+
+        self.model_weights = self.conn.CASTable(name=self.model_name + '_weights')
+
+    def set_weights_attr(self, attr_tbl, clear=True):
+        '''
+        Attach the weights attribute to the model weights
+
+        Parameters
+        ----------
+        attr_tbl : CASTable or string or dict
+            Specifies the CAS table that contains the weights attribute table
+        clear : bool, optional
+            Specifies whether to drop the attribute table after attach it
+            into the weight table.
+
+        '''
+        self._retrieve_('table.attribute',
+                        task='ADD', attrtable=attr_tbl,
+                        **self.model_weights.to_table_params())
+
+        if clear:
+            self._retrieve_('table.droptable',
+                            table=attr_tbl)
+
+        print('NOTE: Model attributes attached successfully!')
+
+    def load_weights_attr(self, path):
+        '''
+        Load the weights attribute form a sashdat file
+
+        Parameters
+        ----------
+        path : string
+            Specifies the server-side directory of the file that
+            contains the weight attribute table.
+
+        '''
+        dir_name, file_name = os.path.split(path)
+        try:
+            flag, cas_lib_name = check_caslib(self.conn, dir_name)
+        except:
+            flag = False
+            cas_lib_name = random_name('Caslib', 6)
+            self._retrieve_('table.addcaslib',
+                            name=cas_lib_name, path=dir_name,
+                            activeOnAdd=False, dataSource=dict(srcType='DNFS'))
+
+        self._retrieve_('table.loadtable',
+                        caslib=cas_lib_name,
+                        path=file_name,
+                        casout=dict(replace=True,
+                                    name=self.model_name + '_weights_attr'))
+
+        self.set_weights_attr(self.model_name + '_weights_attr')
+
+        if not flag:
+            self._retrieve_('table.dropcaslib', caslib=cas_lib_name)
+
+    def save_to_astore(self, path = None, **kwargs):
+        """
+        Save the model to an astore object, and write it into a file.
+
+        Parameters
+        ----------
+        path : string
+            Specifies the client-side path to store the model astore.
+            The path format should be consistent with the system of the client.
+
+        """
+        if not self.conn.queryactionset('astore')['astore']:
```

Comment:
This is another case where I think you'll get a KeyError if the actionset isn't already loaded.